client/modules/accounts/containers/auth/authContainer.js

```diff
@@ -0,0 +1,205 @@
+  }
+
+  handleFormSubmit = (event, email, password) => {
+    event.preventDefault();
+
+    this.setState({
+      isLoading: true
+    });
+    const errors = {};
+    const username = email.trim();
+    const pword = password.trim();
+
+    const validatedEmail = LoginFormValidation.email(username);
+    const validatedPassword = LoginFormValidation.password(pword, { validationLevel: "exists" });
+
+    if (validatedEmail !== true) {
+      errors.email = validatedEmail;
+    }
+
+    if (validatedPassword !== true) {
+      errors.password = validatedPassword;
+    }
+
+    if (_.isEmpty(errors) === false) {
+      this.setState({
+        isLoading: false,
+        formMessages: {
+          errors: errors
+        }
+      });
+      return;
+    }
+
+    if (this.props.currentView === "loginFormSignInView") {
+      Meteor.loginWithPassword(username, pword, (error) => {
+        if (error) {
+          this.setState({
+            isLoading: false,
+            formMessages: {
+              alerts: [error]
+            }
+          });
+        }
+      });
+    } else if (this.props.currentView === "loginFormSignUpView") {
+      const newUserData = {
+        email: username,
+        password: pword
+      };
+
+      Accounts.createUser(newUserData, (error) => {
+        if (error) {
+          this.setState({
+            isLoading: false,
+            formMessages: {
+              alerts: [error]
+            }
+          });
+        }
+      });
+    }
+  }
+
+  hasError = (error) => {
+    // True here means the field is valid
+    // We're checking if theres some other message to display
+    if (error !== true && typeof error !== "undefined") {
+      return true;
+    }
+
+    return false;
+  }
+
+  formMessages = () => {
+    return (
+      <MessagesContainer
+        messages={this.state.formMessages}
+      />
+    );
+  }
+
+  services = () => {
+    const serviceHelper = new ServiceConfigHelper();
+    return serviceHelper.services();
+  }
+
+  shouldShowSeperator = () => {
+    const serviceHelper = new ServiceConfigHelper();
+    const services = serviceHelper.services();
+    const enabledServices = _.filter(services, {
+      enabled: true
+    });
+
+    return !!Package["accounts-password"] && enabledServices.length > 0;
+  }
+
+  capitalize = (str) => {
+    const finalString = str === null ? "" : String(str);
+    return finalString.charAt(0).toUpperCase() + finalString.slice(1);
+  }
+
+  handleSocialLogin = (value) => {
+    let serviceName = value;
+
+    // Get proper service name
+    if (serviceName === "meteor-developer") {
+      serviceName = "MeteorDeveloperAccount";
+    } else {
+      serviceName = this.capitalize(serviceName);
+    }
+
+    const loginWithService = Meteor["loginWith" + serviceName];
+    const options = {}; // use default scope unless specified
+
+    loginWithService(options, () => {
+      // TODO: add error message for failed login attempt
```

Comment:
Update on this: I'll just add an error message here because I've noted an error/issue that happens when logging in with Facebook. This also happens in development. Will open a separate issue for it.